### ISSUE TITLE
wp crash after wrong phpbb path

### ISSUE THREAD

**Issue by RARgames** [Wardormeur/wpphpbbunicorn#5]
When I write wrong path in "phpbb path" Wrong path was //forum/ my site crashed:

Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 167
Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 167

I had to manually delete one of the "/" in my database. This was no problem for me but maybe it would be a problem for some ppl.

**Comment by Wardormeur** [comment]
I may have done it only for one of the 2 paths, will check tomorrow, thanks for the feedback :)
Other solution would have been to delete the cache files (inc/cache/*) in order to regenerate manually by saving again the parameters

**Comment by Wardormeur** [comment]
Should be fixed

**Comment by RARgames** [comment]
Similar error after wrong path
Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172
Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

**Comment by Wardormeur** [comment]
Validation (and refrrsh of yhe cached files) is done upon saving the parameters, did you do it?

**Comment by RARgames** [comment]
I deleted all plugin files(not database parameters set by uniorn)

**Comment by Wardormeur** [comment]
What the error says it that the files that are cached upon settings validation are not set.
The only reasons for it not to be set are : 
- it crashed upon validation
- validation haven't been done

Would you have a more complete Apache log so we see if something happend during the settings validation ?

**Comment by Wardormeur** [comment]
Just updated the plugin with more check, feel free to five it a try

**Comment by RARgames** [comment]
now it's working but... when I save settings 2 times/or just spam the save button I get error:(just minor issue) like above

**Comment by Wardormeur** [comment]
err, the above error isn't minor, is it the same ?

**Comment by RARgames** [comment]
Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

**Comment by Wardormeur** [comment]
It means the cache haven't been created, which is done when you save the parameters. When you save your paths in settings, is there any error message ?

**Comment by RARgames** [comment]
no error message

**Comment by Wardormeur** [comment]
Did you change the default path to Wordpress ? like to "mywp-content" or something ?
Could you describe me the architecture of your server and show me the path you actually set ?

**Comment by RARgames** [comment]
ok,
I have web server. There are many websites on this server so there is my wp folder and it;s separated(so u(browser/wp/etc.) can't go upper folder) so wp folder is 4ga and there is folder forum in it(phpbb directory) I don't know is this path is correct but I tried ( forum, /forum, /forum/) plugin only accepted "forum" as a good path

EDIT

**Comment by RARgames** [comment]
I don't know if the path in phpbb is ok=I set it to none right now(I tried with ../)

**Comment by Wardormeur** [comment]
Hum, havent tried this scenario yet, but it should be /wp/forum if it's a linux base, else it would be drive:/pathtoserverwww/wp/forum

**Comment by RARgames** [comment]
It's linux, /4ga/forum not working:The file config.php does not exists in the path you have enter
/forum not working:The file config.php does not exists in the path you have enter
full path from main: not working (up)
/forum/ - no error but after save spam: 

Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172
Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

**Comment by Wardormeur** [comment]
No need to save spam, if it shows that the forum path is invalid, it's because it is :D
I'll display the absolute path of wp in this page so that it can help to determinate the phpbb one

**Comment by RARgames** [comment]
yeah :P 
so if those 2 path are good everything should work now

! more problem after the path in wp is right (/forum/) everything seems fine but when I switch pages few times(for exemple to users, find someone etc.) the site crashes with the error:

Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

**Comment by Wardormeur** [comment]
Just pushed it, you normally only have to add /yourforumfolder from the given path on the settings as it displays the wordpress path (and your forum is inside wordpress folder so..)
What you have only shows that the checks are good and that the plugin doesnt make the whole website crash anymore when it's misconfigured :D

**Comment by RARgames** [comment]
still the same problem, I save new path it was ok(/forum/) and than go back to main unicorn page(not settings) and:

Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

**Comment by RARgames** [comment]
Maybe I am doing something wrong when I delete old version of plugin?
I delete wp-content/plugins/wpphpbbunicorn only

**Comment by Wardormeur** [comment]
In Settings, there is now a sentence indicating the fullpath of wordpress, you need to add your forum path to it
"This wordpress absolute path is => D:/Dev/UniServerZ/www/wp-4.2.4/wordpress/"
so you fullpath is what's is displayed+/yourforumfolder 
Edit :  it may be displayed as ok as it's relative, but absolute path is necessary watsoever :)

**Comment by RARgames** [comment]
I get it so my path is //forum/ and it saves correctly but when I go to any other page after saving, wp crashes: 

Warning: require_once(/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php): failed to open stream: No such file or directory in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172
Fatal error: require_once(): Failed opening required '/wp-content/plugins/wpphpbbunicorn\inc\cache\common.php' (include_path='/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-text-template:/wp-content/plugins/wpphpbbunicorn/vendor/phpunit/php-timer:.:/usr/local/php/pear5') in /wp-content/plugins/wpphpbbunicorn/wpphpbbunicorn.php on line 172

EDIT: I think the path "//forum/" is good: "This wordpress absolute path is => // "

**Comment by Wardormeur** [comment]
Yep, //forum/ seems good, even though the // is strange.
I just updated some potential problem on file inclusion, but could you tell me if /wp-content/wpphpbbunicorn/inc/cache/ contains any file? To see if it comes from cache generation or path recognition

**Comment by RARgames** [comment]
after crash it contains 3 files: common.php, function_content.php, function_user.php
Before updating

**Comment by RARgames** [comment]
It's working=no error

**Comment by RARgames** [comment]
YaY I think everything is working now!

One final question :D 
All users have to register on phpbb site?

When I log in on a normal user everything is ok, but when I try to log in as admin it log in and give phpbb redirect error:(translate from polish): General error
They tried to redirect to potentially hazardous address.

It's only when I go to my main site 4ga.pl/wp-admin and than plugin redirects me to phpbb site, I log in and this error appears, When I go directly yo the phpbb site it works fine

**Comment by RARgames** [comment]
There is also error in forum publication on wp site: 
Fatal error: Call to a member function fetch_assoc() on a non-object in /wp-content/plugins/wpphpbbunicorn/inc/Forum.php on line 12

and there is no email sync

**Comment by Wardormeur** [comment]
Good, we finally did it :)
Yes, users must be registered into phpbb. Do you have an Firewall or an Antivirus running? Any WP plugin? First time i see this hazardous thing.
Regarding your error, do you have forums on your forums? Can you try to open an article to see if you have the forum list where to publish under the article ?

**Comment by RARgames** [comment]
1)What about email sync, it is not implemented yet or you don't want to do this?

2)Is it possible to log in using login widget on wp site?

I don't know anything about any special antivirus or firewall.

I have no forums list.(I have forums in forums)

**Comment by Wardormeur** [comment]
2) yes, it already exists
1) not sure what you mean, emails are copied from phpbb in creation process of the wp user

**Comment by RARgames** [comment]
so my wp doesn't have any new user email

EDIT: Some wp bug, emails are working

**Comment by RARgames** [comment]
I set default role:"subscriber" in phpbbwp ext, wp is giving role author to every new user

**Comment by Wardormeur** [comment]
This is related to phpbbwpu, not wpphpbb. Try to split a bit please :)

**Comment by RARgames** [comment]
ok, I'll try :D

**Comment by RARgames** [comment]
I had one firewall wp plugin, I disabled it and the problem with hazardous thing still exists

**Comment by Wardormeur** [comment]
Try another browser, in case of ? 
Try deactivate your AV ?

**Comment by RARgames** [comment]
And one big issue now, my wp post editor is not working good.
1)text is invisible(white)
2)visual editor not working at all

I deactivated unicorn plugin for a while and everything is working normally

**Comment by RARgames** [comment]
"Try another browser, in case of ?
Try deactivate your AV ?"

I tried Firefox, Chrome and Opera, it's all the same
I have av deactivated right now, nothing changes

**Comment by Wardormeur** [comment]
Is WP post editor an wp ext ?
Never seen the "unsecure URL" thingy, could i see a picture cause I can't really.. know what happens.

**Comment by RARgames** [comment]
I use TinyMCE Advanced but when I disable it, there is no change
I can send you pic(it's in polish) I connect to 4ga.pl/wp-admin/ it redirects to phpbb login page, when I log there is:
![login](https://cloud.githubusercontent.com/assets/13639766/9155870/77571d0e-3ec7-11e5-9de2-ae15f0fc8658.png)

**Comment by Wardormeur** [comment]
Oh, so this is phpbb related, not WP. You may have an phpbb extension doing that (or a configuration) cause i don't have it on a clean install

**Comment by RARgames** [comment]
I don't have any other ext(only unicorn)

**Comment by RARgames** [comment]
do you know what I can change in setting to make this work?
and what about problem with visual editor?

**Comment by RARgames** [comment]
this error with hazardous thingy is in en more like this:

General Error
Tried to redirect to potentially insecure url.

**Comment by Wardormeur** [comment]
I have sincerely no idea about that. After a quick lookup on google, the reasons seems to be various, like database hours or configuration to generic corruption of files
I think we should give the phpbb forum a try for this particular error

**Comment by RARgames** [comment]
so if a user edit his profile on wp, the changes won't affect user profile on phpbb(use /wp-admin) Is there any way to disable editing profile by unicorn plugin? Adding this option will be great for users(no misunderstanding the site etc.)

Did you find something that can affect on visual editor?

How you want to do this?(I think we should give the phpbb forum a try for this particular error)

**Comment by Wardormeur** [comment]
A change hover WP won't be reflected to phpbb, but a change to phpbb will be reflected to WP ; i can make it optionnal post-creation.
No, didn't look more hover the rest, a link to your editor would be great ;)

**Comment by Wardormeur** [comment]
What version of phpbb 3.1 are you running ?
https://github.com/PayBas/QuickStyle/issues/5

**Comment by RARgames** [comment]
3.1.5

**Comment by RARgames** [comment]
Tiny MCE advannced editor (when I disable it and use default wp editor, the problem still exist.)
wp page:  https://wordpress.org/plugins/tinymce-advanced/
plugin page:  http://www.laptoptips.ca/projects/tinymce-advanced/

**Comment by Wardormeur** [comment]
ty, will have a look at it

**Comment by RARgames** [comment]
It's just a quick list for us to don't forget about something

To sum up:
WP related issues:
1)Visual editor+white text+some buttons like publish don't work
2)forum publication
3)option to block profile

Phpbb related issues:
1)insecure url

**Comment by Wardormeur** [comment]
Hum, that's why we normally do separate issues, this is a real mess to find the info back..
3), Option, i still dont get your point as it's only updated from phpbb, not from WP
2) I need more info : did you checked any forum where to publish the artcile? Is this forum particularly closed to a specific member or anything?
1) & 4) on the go, asap

**Comment by RARgames** [comment]
3) when a member will force editing his profile on wp( using /wp-admin) he will say "why my profile is not updating" It's just that kind of thing

**Comment by Wardormeur** [comment]
3) Well, i dont plan to support editing of WP users to synchronize back to phpbb soon, as i don't see real use-case of data which is handled WP side
1) I can publish and create a new article w/o problem, could you try on a brand new WP w/ only both extensions?

**Comment by RARgames** [comment]
3) ok, it;s not a problem, but is it hard to block editing profile on wp? or just do some kind of info on the top like "Changes won't be saved"

**Comment by RARgames** [comment]
1)Maybe I will create user for you and you will see the editor(what change when you disable unicorn)
I can create new wordpress, but it have to be totally new or I can just disable all plugins?

**Comment by Wardormeur** [comment]
A new one would be better, but an access can also be a good idea :)

**Comment by RARgames** [comment]
Can you send me email with your ip adress? (I have to add you to whitelist on wordpress)
And I will send you back username and password

My email: sswwqqaa@gmail.com

**Comment by RARgames** [comment]
sometimes on wp site something like this appears:

[phpBB Debug] PHP Warning: in file /wp-includes/option.php on line 772: Cannot modify header information - headers already sent by (output started at /wp-includes/functions.php:1378
[phpBB Debug] PHP Warning: in file /wp-includes/option.php on line 773: Cannot modify header information - headers already sent by (output started at /wp-includes/functions.php:1378

What can cause this warning? It appears mostly when using chrome, I had it only once

**Comment by Wardormeur** [comment]
It looks like some echo or headers added post- rendering. It could be a Wordpress pb which is only displayed because your phpbb settings are set to "DEBUG", and now that phpbb is somehow "included" into your WP, it justs display those hidden warnings from WP core/plugins
I just sent you my ip ;)

**Comment by RARgames** [comment]
I sent you mail back with login, password and site url.'
Maybe ipv6 works but I'm not sure. So if you have any problems accessing 4ga.pl(main site) write to me

**Comment by Wardormeur** [comment]
Fixed, thanks for the info :)

**Comment by RARgames** [comment]
Idk what I just did, but the problem with visual editor seems to be resolved
and post publication works too

**Comment by RARgames** [comment]
and post publication works too

**Comment by Wardormeur** [comment]
I fixed it live on your system ;) That's how I've been able to find about the error and how to fix it

**Comment by RARgames** [comment]
ahhh ok, I'm come back, switch some settings and say woooah editor is working :D

**Comment by RARgames** [comment]
Does phpbbwpunicorn sync automatically after adding user to group? or I have to resync every user in options?

**Comment by Wardormeur** [comment]
Should be tested, but considering the code, it should

**Comment by RARgames** [comment]
yeah it works now, it had problems with old (wp) users

If I use widget the problem with hazardous thingy is gone.
Half of login widget is broken for me:
Forum admin link not work (it redirects to) mysite.pl/adm/?sid=263590ea6d30c6f0ae9730a631577900
and I think it should redirect to mysite.pl/forum/adm/?sid=263590ea6d30c6f0ae9730a631577900

Is it possible to set redirect after login to main site?

**Comment by Wardormeur** [comment]
forum link from what you describe seems broken, please open a separate issue if possible ;)
Redirection should be done after login

**Comment by RARgames** [comment]
Created,
Thanks for help
